src/afr-vexu-lib/base-readable/adi_digital_readable.cpp

```diff
@@ -0,0 +1,20 @@
+//
+// Created by Kyle on 11/16/2018.
+//
+
+#include "afr-vexu-lib/base-readable/adi_digital_readable.h"
+#include "afr-vexu-lib/defines.h"
+
+AFR::VexU::error_t AFR::VexU::BaseReadable::adi_digital_readable::update_private(const double& delta_seconds){
+    int32_t temp_val = digital.get_value();
+    AFR_PROS_INTERNAL_CALL(temp_val, PROS_ERR);
+    if(temp_val == 1) readable::value = true;
```

Comment:
In c++ bool is a number with true being 1 and false being 0. You can cast and make this faster and more concise.